packages/api/endpoints/granules.js

```diff
@@ -940,6 +949,146 @@ async function del(req, res) {
   return res.send({ detail: 'Record deleted', ...deletionDetails });
 }
 
+const bulkMoveCollectionSchema = z.object({
+  sourceCollectionId: z.string().nonempty('sourceCollectionId is required'),
+  targetCollectionId: z.string().nonempty('targetCollectionId is required'),
+  batchSize: z.number().positive().optional().default(100),
+  concurrency: z.number().positive().optional().default(100),
+  invalidBehavior: z.enum(['error', 'skip']).default('error'),
+  cmrGranuleUrlType: z.enum(['http', 's3', 'both']).default('both'),
+  s3MultipartChunkSize: z.number().optional(),
+  executionName: z.string().optional(),
+});
+const parseBulkMoveCollectionPayload = zodParser('BulkMoveCollection payload', bulkMoveCollectionSchema);
+
+/**
+ * Bulk move granules to a new collection.
+ *
+ * @param {object} req - The request object.
+ * @param {object} req.body - The request payload.
+ * @param {string} req.body.sourceCollectionId - The source collection ID.
+ * @param {string} req.body.targetCollectionId - The target collection ID.
+ * @param {number} [req.body.batchSize=100] - The batch size for processing granules.
+ * @param {number} [req.body.concurrency=100] - The concurrency level for processing granules.
+ * @param {string} [req.body.invalidBehavior='error'] - The behavior for invalid granules
+ * ('error' or 'skip').
+ * @param {number} [req.body.s3MultipartChunkSize] - The S3 multipart chunk size.
+ * @param {string} [req.body.executionName] - The execution name.
+ * @param {object} testContext - The test context object
+ * @param {object} res - The response object.
+ * @returns {Promise<Object>} The response object with the execution ARN and message.
+ */
+async function bulkMoveCollection(req, res) {
+  const {
+    knex = await getKnexClient(),
+    sfnMethod = sfn,
+    workflow = 'HelloWorldWorkflow',
+  } = req.testContext || {};
+
+  const collectionPgModel = new CollectionPgModel();
+  const granulePgModel = new GranulePgModel();
+
+  const body = parseBulkMoveCollectionPayload(req.body);
+  if (isError(body)) {
+    return returnCustomValidationErrors(res, body);
+  }
+
+  if (!process.env.system_bucket) {
+    return res.boom.badRequest('API is misconfigured, system_bucket must be defined in the env variables');
+  }
+  if (!process.env.stackName) {
+    return res.boom.badRequest('API is misconfigured, stackName must be defined in the env variables');
+  }
+  //get collection
+  const pgCollection = await collectionPgModel.get(
+    knex,
+    deconstructCollectionId(body.sourceCollectionId)
+  );
+  // check collection returned
+  if (pgCollection === undefined) {
```

Comment:
I believe the BaseModel `get` will throw, not return undefined?

Reply:
Great catch - this was a leftover from an earlier implementation approach. 

469cffe5481